pkg/providers/aws/provider_postgres.go

```diff
@@ -700,11 +700,12 @@ func buildRDSUpdateStrategy(rdsConfig *rds.CreateDBInstanceInput, foundConfig *r
 			return nil, errorUtil.Wrap(err, "invalid postgres version")
 		}
 		if engineUpgradeNeeded {
+			logrus.Info(fmt.Sprintf("Engine upgrade found, the current EngineVersion is %s and is upgrading to %s", *foundConfig.EngineVersion, *rdsConfig.EngineVersion))
 			mi.EngineVersion = rdsConfig.EngineVersion
 			if cr.Spec.ApplyImmediately {
 				mi.ApplyImmediately = aws.Bool(cr.Spec.ApplyImmediately)
 			}
-			updateFound = true
+			return mi, nil
 		}
 	}
 	if !updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues) {
```

Comment:
```suggestion
			updateFound = true
		}
	}
	if (!updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues)) && !mi.ApplyImmediately  {
```

sorry for all the updates @austincunningham but I believe this is the way to go. :)

Reply:
Feel free to give me a shout and we can chat through it.

Reply:
was going with 
```go
	if mi.EngineVersion != nil && pm.EngineVersion != nil {
		if *mi.EngineVersion == *pm.EngineVersion {
			pendingModifications = false
		}
		if cr.Spec.ApplyImmediately {
			logrus.Infof(fmt.Sprintf("Engine version pending modification changeing %s version to %s", *pm.EngineVersion, *mi.EngineVersion))
			pendingModifications = true
		}
	}
```
but all would work, not invested in any way in particular

Reply:
Looks like that trows an nil pointer on the unit tests, if your happy with
```go
if (!updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues)) && mi.ApplyImmediately != nil  {
```